unikernel.ml

```diff
@@ -973,62 +995,189 @@ struct
+                     })
+                ~icon:"/images/robur.png" ())
+             `Internal_server_error)
+    | Some (unikernel_name, unikernel) -> (
+        (Albatross.query_console ~domain:user.name albatross ~name >|= function
+         | Error err ->
+             Logs.warn (fun m ->
+                 m "error querying console of albatross: %s" err);
+             []
+         | Ok (_, console_output) -> console_output)
+        >>= fun console_output ->
+        let now = Ptime.v (P.now_d_ps ()) in
+        generate_csrf_token store user now reqd >>= function
+        | Ok csrf ->
+            Lwt.return
+              (reply reqd ~content_type:"text/html"
+                 (Dashboard.dashboard_layout ~csrf user
+                    ~content:
+                      (Unikernel_single.unikernel_single_layout
+                         (unikernel_name, unikernel)
+                         now console_output)
+                    ~icon:"/images/robur.png" ())
+                 ~header_list:[ ("X-MOLLY-CSRF", csrf) ]
+                 `OK)
+        | Error err ->
+            Lwt.return
+              (reply reqd ~content_type:"text/html"
+                 (Guest_layout.guest_layout ~page_title:"500 | Mollymawk"
+                    ~content:(Error_page.error_layout err)
+                    ~icon:"/images/robur.png" ())
+                 `Internal_server_error))
+
+  let unikernel_prepare_update albatross store name reqd http_client
+      ~json_dict:_ (user : User_model.user) =
+    (* TODO use uuid in the future *)
+    user_unikernel albatross ~user_name:user.name ~unikernel_name:name
+    >>= fun unikernel_info ->
+    match unikernel_info with
+    | None ->
+        Middleware.redirect_to_error ~data:(`String "Builder_web request error")
+          ~title:("An error occured while fetching " ^ name)
+          ~api_meth:false `Internal_server_error reqd ()
+    | Some (unikernel_name, unikernel) -> (
+        Builder_web.send_request http_client
+          (Builder_web.base_url ^ "/hash?sha256=" ^ Ohex.encode unikernel.digest)
+        >>= function
+        | Error (`Msg err) ->
+            Logs.err (fun m -> m "Builder_web update error %s" err);
+            Middleware.redirect_to_error
+              ~data:(`String ("Builder_web request: " ^ err))
+              ~title:(Vmm_core.Name.to_string unikernel_name ^ " update Error")
+              ~api_meth:false `Internal_server_error reqd ()
+        | Ok response_body -> (
+            match
+              Builder_web.build_of_json (Yojson.Basic.from_string response_body)
+            with
+            | Error (`Msg err) ->
+                Logs.err (fun m -> m "Builder_web update error %s" err);
+                Middleware.redirect_to_error
+                  ~data:(`String ("Builder_web request: " ^ err))
+                  ~title:
+                    (Vmm_core.Name.to_string unikernel_name ^ " update Error")
```

Comment:
same as above. Maybe give some hints what failed. After all, having distinct error messages in distinct code paths helps us developers to figure out where the issue is.